core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -73,7 +73,10 @@ class HadoopMapReduceCommitProtocol(jobId: String, path: String)
 
     val stagingDir: String = committer match {
       // For FileOutputCommitter it has its own staging path called "work path".
-      case f: FileOutputCommitter => Option(f.getWorkPath.toString).getOrElse(path)
+      case f: FileOutputCommitter =>
+        val workPath = f.getWorkPath
+        require(workPath != null, s"Committer has no workpath $f")
+        Option(workPath.toString).getOrElse(path)
```

Comment:
If this was intended to be resilient, it'd have to become `Option(workpath).getOrElse(path).toString()`, though that would actually hide the problem I'd managed to create: hand in a job committer to a task.